### DISCUSSION TITLE
Command line tool for question manipulation

### DISCUSSION THREAD

**Discussion by trombonekenny** [PrairieLearn/PrairieLearn#9477] [Ideas]
Would there be interest in a command line tool (presumably in the container, or flexible enough to be installed locally without too much trouble) to do some common PL course content manipulation from the shell?

Some ideas:
- create a new question folder structure
- copy a question
- move a question between folders (and update all references)
- launch the external grader locally for a question, maybe with test or mockup question data, or a shell?
- confirm every question folder passed in has a specific set of info.json configuration set (or force it)

What else would people use?

**Comment by mwest1066** [comment]
My general thoughts on this are that we'd get more utility from building better online tools into the web UI. For example, a "mass rename questions" feature that takes a CSV with `old_QID,new_QID` columns, or tools to make bulk changes to question properties.

However, I do like the idea of a harness for running an external autograder locally without PL at all.

**Reply by eliotwrobson** [reply]
I agree that people will get more use out of something built directly into the web UI, but having a command line tool might be a good intermediate step. A simple tool like this would allow us to gain a good perspective on potential use cases, and would be substantially less developer overhead than going directly for something integrated into the site.
